packages/plugins/src/hardhat/simulate.ts

```diff
@@ -303,25 +314,61 @@ export const simulateDeploymentSubtask = async (
     removeRoles,
     wallet: signer,
   }
-  const result = await compileAndExecuteDeployment(simulationContext)
 
-  if (!result) {
-    throw new Error(
-      'Simulation failed for an unexpected reason. This is a bug. Please report it to the developers.'
-    )
-  }
+  let attempts = 0
+  while (executionCompleted === false) {
+    try {
+      const result = await callWithTimeout(
+        compileAndExecuteDeployment(simulationContext),
+        90000,
```

Comment:
is 90 seconds a safe value? i'm trying to think of the worst case scenario that could cause this call to prematurely timeout. maybe the worst case is that the user has a very large deployment on a single chain w/ a free tier (i.e. slow) provider. i'm not opinionated, but curious to hear your thoughts

Reply:
I think you're looking at this incorrectly. We have to operate under the assumption that this call will inevitably time out in cases where it should not. For example, a rootstock deployment via the website will generally always timeout since the network is so slow. We would have to use a high timeout period to avoid that (like 10 minutes or more). Furthermore, even if we use a very high timeout period a user will inevitably come around with a deployment that is so large that the execution time exceeds the timeout period we set. We cannot prevent this from happening.

Since this function will inevitably timeout, we must design it so that it doesn't matter if it times out unnecessarily. The next attempt to execute the `compileAndExecuteDeployment` function will flawlessly recover no matter what. I've put a lot of work into ensuring this is the case. This strategy has proven to be very reliable on the website already. We've had lots of bugs but never related to this. 

It's worth noting that even if we could ensure that the function would never ever timeout, we would still need to design the `compileAndExecuteDeployment` to recover in all situations because our backend server will inevitably go down and some point, and we need to be able to recover from that. 

So the right way to look at the timeout is to think about what a reasonable time period is for timing out in cases where we legitimately want a timeout to occur (such as an rpc provider stalling, a transaction failing to get executed, etc). Then we should set a timeout period that is appropriate for how long we are willing to wait for that type of situation. 

Honestly, since this simulation is running against a local node that is quite fast (in comparison to a live network) it is very likely that we could reduce this time out further. However, I don't really think there's much value in that. I do not expect the timeout to actually occur very often during the simulation.